composer/loggers/logger_destination.py

```diff
@@ -38,29 +38,40 @@ class LoggerDestination(Callback, ABC):
             Batch 0: {'rank_zero_seed': ...}
     """
 
-    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]):
-        """Log data.
+    def log_hyperparameters(self, hyperparameters: Dict[str, Any]):
+        """Log hyperparameters, configurations, and settings.
 
-        Subclasses should implement this method to store logged data (e.g. write it to a file, send it to a server,
-        etc...). However, not all loggers need to implement this method.
+        Logs any parameter/configuration/setting that doesn't vary during the run.
 
-        .. note::
+         Args:
+            hyperparameters (Dict[str, Any]): A dictionary mapping hyperparameter names
+                (strings) to their values (Any).
+        """
+        del hyperparameters  # unused
+        pass
 
-            This method will block the training loop. For optimal performance, it is recommended to deepcopy the
-            ``data`` (e.g. ``copy.deepcopy(data)``), and store the copied data in queue. Then, either:
+    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
```

Comment:
This new API seems a bit limiting? `wandb.log(data)` can take any type of data (time series, images, etc.) yet now through the logger you can _only_ log scalars. 

e.g. https://github.com/mosaicml/composer/blob/dev/composer/callbacks/image_visualizer.py has not been updated in this PR and would break, correct?

Reply:
ok fixed by changing `log_metrics` to Any for WandB and InMemoryLogger and calling `log_metrics` for the image visualizer